### DISCUSSION TITLE
@timed decorator for debug environments only

### DISCUSSION THREAD

**Discussion by NicoN13** [anvilistas/anvil-extras#529] [Q&A] [answered]
Hi all,

I'm trying to set the @timed decorator to only display for my debug env and not the production env.

This is my attempt so far:

`
import logging
import time

timed_logger = logging.getLogger(__name__)
if "debug" in anvil.app.environment.tags:
  print("DEV instance - logging set to debug.")
  timed_logger.setLevel(logging.DEBUG)
timed_kwargs = {"logger":timed_logger, "level":logging.DEBUG}

@timed(**timed_kwargs)
def my_func():
    print("Started my_func")
    time.sleep(2)
    print("Done with my_func")
`

What I would expect to see with the above code: If I run from the IDE, then it would log the time as per usual when simply using @timed decorator. Then if I run from a prod link, it would not show anything for the timing of calling my_func()

It's not printing the time for either debug or prod environments. Any suggestions where I'm going wrong?

![image](https://github.com/anvilistas/anvil-extras/assets/117717063/b8de5efd-5477-42ac-8df0-e41211c6ad02)

**Comment by s-cork** [comment, marked as answer]
If you want something simple then just use an anvil extras logger instead of Python's logging. 

If you want to use pythons logger you'll need to follow the docs to send the stream to stdout. You should be able to find examples online of how to do this.

**Reply by NicoN13** [reply]
Aah thanks, I tried that originally but it didn't want to import logging as per the docs.

I simply had to change where logging gets imported from. The docs say we should import it like this: 
from anvil_extras.utils import timed, logging

but logging couldn't be found within utils when I tried. The below works though.

![image](https://github.com/anvilistas/anvil-extras/assets/117717063/193520e7-3b37-4833-a034-e886e5256f49)

**Reply by s-cork** [reply]
Ah if there's a bad import in the docs we should fix that. Do you want to PR a fix?

**Reply by NicoN13** [reply]
@s-cork Sure, will do it asap.
